### PR TITLE
T/50: Add `--ignore` option

### LINKED ISSUE #50: Add `--ignore` option
Just like Lerna's `--ignore` https://github.com/lerna/lerna#--ignore-glob

We need it to skip `ckeditor5-build-*` when doing some maintenance tasks on CKEditor 5 packages.

We can also implement `--scope` so e.g. one can do:

```
--scope "package-*" --ignore "package-util-*
```

And I've been also thinking about "groups", defined in `mgit.json` so one can easily pick on which packages the command should be executed without writing the globs every time.

Anyway, let's start with `--ignore` for now and possibly `--scope`.

### PR DESCRIPTION
**Pull request by pomek**
Feature: Introduced new options which allow specifying packages for executing commands. Closes #50.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -9,6 +9,8 @@ const path = require( 'path' );
 const createForkPool = require( './utils/createforkpool' );
 const logDisplay = require( './utils/displaylog' );
 const getOptions = require( './utils/getoptions' );
+const getPackageNames = require( './utils/getpackagenames' );
+const chalk = require( 'chalk' );
```
Comment (ma2ciek) [inline lib/index.js]: duplicated with https://github.com/cksource/mgit2/pull/51/files#diff-6d186b954a58d5bb740f73d84fe39073R114.

File: lib/utils/getoptions.js [hunk 2]
```diff
@@ -23,7 +23,9 @@ module.exports = function cwdResolver( callOptions, cwd ) {
 		resolverPath: path.resolve( __dirname, '../default-resolver.js' ),
 		resolverUrlTemplate: 'git@github.com:${ path }.git',
 		resolverTargetDirectory: 'git',
-		resolverDefaultBranch: 'master'
+		resolverDefaultBranch: 'master',
+		ignore: null,
+		scope: null
 	};
 
 	if ( fs.existsSync( mgitJsonPath ) ) {
```
Comment (ma2ciek) [inline lib/utils/getoptions.js]: There's something wrong with the line below.

Comment (ma2ciek) [inline lib/utils/getoptions.js]: And with [L35](https://github.com/cksource/mgit2/pull/51/files#diff-da67e5c5e86dbd7f7f1844dcab3edeb0R35) as well.

File: lib/utils/getpackagenames.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+const minimatch = require( 'minimatch' );
+
+/**
+ * @param {Options} options
```
Comment (ma2ciek) [inline lib/utils/getpackagenames.js]: The options param in this function doesn't need to satisfy the `Options` interface.

File: lib/index.js [hunk 4]
```diff
@@ -9,6 +9,8 @@ const path = require( 'path' );
 const createForkPool = require( './utils/createforkpool' );
 const logDisplay = require( './utils/displaylog' );
```
Comment (ma2ciek) [inline lib/index.js]: logDisplay = displayLog? :D

File: lib/index.js [hunk 5]
```diff
@@ -43,6 +45,12 @@ module.exports = function( args, options ) {
 		enqueue( item );
 	}
 
+	if ( allPackagesNumber === 0 ) {
```
Comment (ma2ciek) [inline lib/index.js]: Maybe this break could be earlier?

File: lib/index.js (hunk 4 above)
Comment (Reinmar) [inline lib/index.js]: The whole logger business needs some attention and we started improving it in https://github.com/cksource/mgit2/pull/46.

Comment (ma2ciek) [inline lib/index.js]: Big +1 for a `getLogger()` function ;)

File: tests/utils/getpackagenames.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+
+/* jshint mocha:true */
+
+'use strict';
+
+const getPackageNames = require( '../../lib/utils/getpackagenames' );
+const expect = require( 'chai' ).expect;
+
+describe( 'utils', () => {
+	describe( 'getPackageNames()', () => {
```
Comment (Reinmar) [inline tests/utils/getpackagenames.js]: Can you pass multiple rules in a glob string – like – `a,b,c`? Please check if minimatch supports it this way and if so, let's have a test for it. If not, let's ignore this as it's not worth too much time.

File: lib/index.js [hunk 7]
```diff
@@ -43,6 +45,12 @@ module.exports = function( args, options ) {
...
+		console.log( chalk.red( 'Not found packages to execute given command.' ) );
```
Comment (Reinmar) [inline lib/index.js]: No packages found that much the given criteria.

Comment (ma2ciek) [inline lib/index.js]: *match ;)

File: tests/utils/getpackagenames.js (hunk 6 above)
Comment (pomek) [inline tests/utils/getpackagenames.js]: No, you can't.

**Comment by pomek** [comment]
Could we merge this PR? I would like to avoid a similar situation like in #46.

**Comment by Reinmar** [comment]
My only doubt is about requiring you to pass the entire glob like `"*editor*"` if you want to filter out or include anything which contains the `"editor"` bit. I think that we may find it inconvenient, but let's see.

**Pull request merged**